.github/workflows/build-vscode-extension.yml

```diff
@@ -20,50 +20,147 @@ jobs:
         id: checksecret_job
         run: |
           echo "is_SONAR_TOKEN_set=${{ env.SONAR_TOKEN != '' }}" >> $GITHUB_OUTPUT
-  build_extension:
+  
+  test_and_generate_binaries:
     needs: [checksecret]
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        binaries:
+          - architecture: win-x64
+            os: windows-latest
+          - architecture: linux-x64
+            os: ubuntu-latest
+          - architecture: osx-arm64
+            os: macOS-latest    
+    runs-on: ${{ matrix.binaries.os }}
     steps:
       - uses: actions/checkout@v4
+
       - uses: actions/setup-node@v4
         with:
           node-version: 20.x
+
+      - name: Use .NET 9
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '9.x'
+
       - id: last_release
         run: |
           $releaseVersion = gh release list --exclude-drafts --exclude-pre-releases --json tagName,isLatest | ConvertFrom-Json | ? {$_.isLatest -eq $true} | select -first 1 -ExpandProperty tagName
+          $releaseVersion = "$releaseVersion" -replace '^v', ''
           Write-Output "RELEASE_VERSION=$releaseVersion" >> $Env:GITHUB_OUTPUT
         shell: pwsh
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - run: scripts/update-vscode-releases.ps1 -version "${{ steps.last_release.outputs.RELEASE_VERSION }}" -packageJsonFilePath "./vscode/microsoft-kiota/package.json" -runtimeFilePath "./vscode/microsoft-kiota/src/kiotaInterop/runtime.json" -online
+
+      - run: scripts/update-vscode-releases.ps1 -version "v${{ steps.last_release.outputs.RELEASE_VERSION }}" -packageJsonFilePath "./vscode/microsoft-kiota/package.json" -runtimeFilePath "./vscode/npm-package/runtime.json" -online
         shell: pwsh
+        
       - name: Install dependencies
-        run: npm install
-        working-directory: vscode/microsoft-kiota
-      - run: npm run lint
-        working-directory: vscode/microsoft-kiota
-      - name: run tests(linux)
-        if: runner.os == 'Linux'
-        run: xvfb-run -a npm run test:coverage
-        working-directory: vscode/microsoft-kiota
-      - name: run tests
-        if: runner.os != 'Linux'
-        run: npm run test:coverage
-        working-directory: vscode/microsoft-kiota
+        run: |
+          npm install
+          npm list
```

Comment:
why an npm list here?

Reply:
it was there to confirm that the npm package was being symlinked. Not needed now. Removing it